EpikCord/client/http_client.py

```diff
@@ -161,14 +164,30 @@ async def request(self, method: str, url: str, *args,  discord: bool = True, cha
         if url.endswith("/"):
             url = url[:-1]
         
-        bucket = self.buckets.get(url) or MockBucket()
+        bucket = self.buckets.get(f"{method}:{url}") or MockBucket()
 
         await self.global_event.wait()
         await bucket.wait()
 
         async with self.session.request(method, url, *args, **kwargs) as response:
             if isinstance(bucket, MockBucket) and response.headers.get("X-RateLimit-Bucket"):
                 if channel_id or guild_id or webhook_id or webhook_token:
+                    bucket = TopLevelBucket(
+                        major_parameters=clear_none_values({
+                            "channel_id": channel_id,
+                            "guild_id": guild_id,
+                            "webhook_id": webhook_id,
+                            "webhook_token": webhook_token
+                        })
+                    )
+                    self.buckets[f"{method}:{url}"] = bucket
+                else:
+                    bucket = Bucket(bucket_hash=response.headers["X-RateLimit-Bucket"])
+                    if bucket in self.buckets.values():
+                        self.buckets[f"{method}:{url}"] = list(self.buckets.values())[list(self.buckets.values()).index(bucket)]
+                        bucket = self.buckets[f"{method}:{url}"]
+                    else:
+                        self.buckets[f"{method}:{url}"] = bucket
```

Comment:
Extract this to a `get_bucket` function?

Reply:
A `set_bucket`, yes. I will do that soon.